datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -1207,6 +1295,28 @@ public void validateDatastreamsUpdate(List<Datastream> datastreams) throws Datas
     }
   }
 
+  /**
+   * Validate if the partition assignment is supported for the datastream
+   * @param datastream datastream which needs the verification
+   * @throws DatastreamValidationException if partition assignment is not supported
+   */
+  public void validatePartitionAssignmentSupported(Datastream datastream) throws DatastreamValidationException {
+    try {
+      String connectorName = datastream.getConnectorName();
+      ConnectorInfo connectorInfo = _connectors.get(connectorName);
+      if (connectorInfo == null) {
+        throw new DatastreamValidationException("Invalid connector: " + connectorName);
+      }
+      if (!(connectorInfo.getAssignmentStrategy() instanceof StickyPartitionAssignmentStrategy)) {
```

Comment:
I probably need some suggestions on this. One alternative is to add an interface in coordinator to indicate if it is supported, but I'm wondering if there is a better way.

Reply:
Have fixed to use the coordinator.getDatastreamPartitions to verify instead. This can also allows us to verify not only the connector but also individual datastream. Thanks for your suggestion.